modules/client-common/test/schemas.test.ts

```diff
@@ -0,0 +1,951 @@
+        where: TEST_ADDRESS,
+        who: TEST_ADDRESS,
+      };
+      expect(() =>
+        MultiTargetPermissionSchema.strict().validateSync(
+          multiTargetPermission,
+        )
+      ).toThrow(
+        new ValidationError("operation is a required field"),
+      );
+    });
+  });
+
+  describe("Test PrepareInstallation", () => {
+    it("should validate a valid PrepareInstallation", () => {
+      const prepareInstallationParams: PrepareInstallationParams = {
+        daoAddressOrEns: TEST_ADDRESS,
+        pluginRepo: TEST_ADDRESS,
+        version: {
+          release: 1,
+          build: 1,
+        },
+        installationParams: [],
+        installationAbi: [],
+      };
+      PrepareInstallationSchema.strict().validateSync(
+        prepareInstallationParams,
+      );
+    });
+    it("should validate a valid PrepareInstallation without optional params", () => {
+      const prepareInstallationParams: PrepareInstallationParams = {
+        daoAddressOrEns: TEST_ADDRESS,
+        pluginRepo: TEST_ADDRESS,
+      };
+
+      PrepareInstallationSchema.strict().validateSync(
+        prepareInstallationParams,
+      );
+    });
+    it("should throw an error if the daoAddressOrEns is missing", () => {
+      const prepareInstallationParams = {
+        pluginRepo: TEST_ADDRESS,
+      };
+      expect(() =>
+        PrepareInstallationSchema.strict().validateSync(
+          prepareInstallationParams,
+        )
+      ).toThrow(
+        new ValidationError("daoAddressOrEns is a required field"),
+      );
+    });
+    it("should throw an error if the pluginRepo is missing", () => {
+      const prepareInstallationParams = {
+        daoAddressOrEns: TEST_ADDRESS,
+      };
+      expect(() =>
+        PrepareInstallationSchema.strict().validateSync(
+          prepareInstallationParams,
+        )
+      ).toThrow(
+        new ValidationError("pluginRepo is a required field"),
+      );
+    });
+    it("should throw an error if the daoAddressOrEns is invalid", () => {
+      const prepareInstallationParams: PrepareInstallationParams = {
+        daoAddressOrEns: TEST_INVALID_ADDRESS,
+        pluginRepo: TEST_ADDRESS,
+      };
+      expect(() =>
+        PrepareInstallationSchema.strict().validateSync(
+          prepareInstallationParams,
+        )
+      ).toThrow(
+        new ValidationError(new InvalidAddressOrEnsError().message),
+      );
+    });
+    it("should throw an error if the pluginRepo is invalid", () => {
+      const prepareInstallationParams: PrepareInstallationParams = {
+        daoAddressOrEns: TEST_ADDRESS,
+        pluginRepo: TEST_INVALID_ADDRESS,
+      };
+      expect(() =>
+        PrepareInstallationSchema.strict().validateSync(
+          prepareInstallationParams,
+        )
+      ).toThrow(
+        new ValidationError(new InvalidAddressOrEnsError().message),
+      );
+    });
+    it("should throw an error if the version is invalid", () => {
```

Comment:
if you can I would check if the version is a typeof `VersionTagSchema` then you can skip verifying if the version number is invalid because `VersionTagSchema` is tested separately.
If not I would add the same tests as done for `VersionTagSchema` to make sure the validation works properly